### PR TITLE
[new-twitch] [Fix] Bttv Channel Emotes syncing when changing channel.

### PR DESCRIPTION
**Pull request by DayBr3ak**
Currently, changing channel doesn't always update the bttv channel emotes. A refresh is usually needed. This PR fixes this.

### CODE REVIEW

File: src/watcher.js [hunk 1]
```diff
@@ -23,7 +23,12 @@ const loadPredicates = {
         const lastReference = currentChatReference;
         const currentChat = twitch.getCurrentChat();
 
-        if (currentChat && currentChat === lastReference) return false;
+        if (currentChat && currentChat === lastReference) {
+            return false;
+        }
+        if (lastReference && currentChat.props.channelID.toString() === lastReference.props.channelID.toString()) {
```
Comment (night) [inline src/watcher.js]: i actually want it to reload from our api in the case when a user navs to a new page from the same user.. emotes and bots can get updated

File: src/watcher.js [hunk 2]
```diff
@@ -259,7 +264,16 @@ class Watcher extends SafeEventEmitter {
             channel = currentChannel;
             api
                 .get(`channels/${channel.name}`)
-                .then(d => this.emit('channel.updated', d));
+                .then(d => {
```
Comment (night) [inline src/watcher.js]: would prefer these without brackets

File: src/watcher.js [hunk 3]
```diff
@@ -259,7 +264,16 @@ class Watcher extends SafeEventEmitter {
...
+                    this.emit('channel.updated', d);
+                })
+                .catch(error => {
+                    this.emit('channel.updated', {
+                        bots: [],
+                        emotes: [],
+                        status: error.status ? error.status : 404
```
Comment (night) [inline src/watcher.js]: status should be 0 if unknown.. you can also `error.status || 0`

File: src/watcher.js [hunk 4]
```diff
@@ -255,14 +255,19 @@ class Watcher extends SafeEventEmitter {
             const currentChannel = twitch.getCurrentChannel();
             if (!currentChannel) return;
 
-            if (currentChannel.id === channel.id) return;
+            if (currentChannel.name === channel.name) return;
```
Comment (DayBr3ak) [inline src/watcher.js]: The name is updated with the `load.chat` event. It's not always the case for the id.

**Comment by DayBr3ak** [comment]
If this is not suitable, maybe: Actually wait for the channel id to be updated (like the watcher `waitFor` does). Or: subscribe to the graphql client and watch channel updates requests?

File: src/watcher.js (hunk 4 above)
Comment (night) [reply src/watcher.js]: im not sure why this was changed or why it's needed

**Comment by DayBr3ak** [comment]
@night this issue is still present, the channel ID is not updated when you switch channels. Bttv emotes are then not updated.

**Comment by night** [comment]
Unfortunately I cannot reproduce that issue. If you have reproduction steps make an issue/PR with steps so I can test.

**Pull request merged**